### DISCUSSION TITLE
How to work with an extension that provides the LS?

### DISCUSSION THREAD

**Discussion by invm** [TypeFox/monaco-languageclient#693] [Q&A] [answered]
Hello again,

I'm trying to work with an extension which provides syntax highlight and lsp for terraform files:
<img width="762" alt="image" src="https://github.com/TypeFox/monaco-languageclient/assets/52990657/015bbf00-26e0-4183-aac8-483e691bc354">
[link](https://marketplace.visualstudio.com/items?itemName=HashiCorp.terraform)

I've downloaded it like in the github theme example and able to import it and even get part of it working:
I get syntax highlight
<img width="457" alt="image" src="https://github.com/TypeFox/monaco-languageclient/assets/52990657/75e1df86-f5e6-422b-997f-817835b9d257">

But the lsp does not start, or at least not working and I'm not seeing in the htop like I see it when it runs from vscode.

I think I'm doing something wrong in this part:
<img width="670" alt="image" src="https://github.com/TypeFox/monaco-languageclient/assets/52990657/0e2095b9-6800-4545-822c-334638657004">

I configure the editor the same like in the python example:
```js
export const runTerraformWrapper = async () => {
  const mainTfUri = vscode.Uri.file('/workspace/main.tf');
  const resourcesTfUri = vscode.Uri.file('/workspace/resources.tf');

  const fileSystemProvider = new RegisteredFileSystemProvider(false);
  fileSystemProvider.registerFile(
    new RegisteredMemoryFile(mainTfUri, mainTfCode),
  );
  fileSystemProvider.registerFile(
    new RegisteredMemoryFile(resourcesTfUri, resourcesTfCode),
  );

  registerFileSystemOverlay(1, fileSystemProvider);
  const userConfig = createUserConfig(
    '/workspace',
    resourcesTfCode,
    resourcesTfUri.toString(),
  );
  const htmlElement = document.getElementById('monaco-editor-root');
  const wrapper = new MonacoEditorLanguageClientWrapper();

  try {
    document
      .querySelector('#button-start')
      ?.addEventListener('click', async () => {
        if (wrapper.isStarted()) {
          console.warn('Editor was already started!');
        } else {
          await wrapper.init(userConfig);

          // open files, so the LS can pick it up
          await vscode.workspace.openTextDocument(resourcesTfUri);
          await vscode.workspace.openTextDocument(mainTfUri);

          await wrapper.start(htmlElement);
        }
      });
    document
      .querySelector('#button-dispose')
      ?.addEventListener('click', async () => {
        await wrapper.dispose();
      });
  } catch (e) {
    console.error(e);
  }
};

```

Would appreciate any help on making the lsp work.

**Comment by CGNonofr** [comment]
Did you manage to make it work in vscode.dev?

> But the lsp does not start, or at least not working and I'm not seeing in the htop like I see it when it runs from vscode.

You're in the browser, it's not able to start new process. The extension needs to be browser-compatible. It seems it is (I was able to install it in vscode.dev) but I don't know how it impacts features. I wasn't able to display any intellisense but I don't know what can be expected.

**Comment by invm** [comment]
Thank you for that clarification, my required functionality does not work in vscode.dev in that extension, only syntax highlight.

I want to use this package in a desktop app, with similar functionality to vscode, I initially wanted to use monaco-vscode-api but cannot download that extension there. 

How would one create a desktop app with this package?

Maybe you're familiar with what gives vscode the ability to open processes?

**Reply by CGNonofr** [reply, marked as answer]
Unfortunately, the code bundled by monaco-vscode-api (used by monaco-languageclient) is the one designed for the web version of VSCode, so it doesn't include the node extension host. It would be a nice feature to be able to use the code designed for the native version, but it's not done yet. And since I don't have that needs, it's not really planned short-term

What you can probably do is using the server extension host. You have to run the VSCode server, and connect to it. You can have a look at https://github.com/CodinGame/monaco-vscode-api/blob/main/docs/vscode_server.md

**Reply by invm** [reply]
You, sir, are a gem.

Thank you.
